grpc-client-server/README.md

```diff
@@ -0,0 +1,188 @@
+# GRPC Client and Server Communication 
+
+The goals of this project are: 
+ * to use gRPC requests, and understand how they differ from REST
+ * to use deadlines with gRPCs
+ * to build a simple prober
+ * optionally, gain experience with instrumenting an application and running Prometheus
+
+## What is gRPC?
+
+RPCs aren’t the same as HTTP requests. HTTP is a text-based protocol, oriented towards resources or entities. RPCs are oriented towards calling a specific function. RPCs are highly structured; normally the request and response are encoded in efficient binary formats which are not human-readable. HTTP APIs are usually limited to CRUD (Create, Read, Update, Delete) operations and RPCs can perform any kind of operation. If you want to efficiently integrate two systems that you control, RPCs are a good choice. If you want to provide an API for use by developers outside of your organisation then HTTP APIs are generally a better choice, because they are simpler to develop against and all programming languages provide good HTTP support. gRPC, which we will use in this exercise, is a RPC implementation that is fairly common in industry. 
```

Comment:
I think this paragraph is confusing if the distinction between (for example) REST, HTTP, protocols, architectural patterns and specific implementations isn't clear — which I think it won't be for the trainees. They may not be aware they've been doing diet-REST and certainly we haven't introduced it. I suggest that, if we want to compare to REST, we need to give an introduction. Otherwise, I would just say something in the direction of:

* We've used simple HTTP request so far, making and handling `GET` and `POST` requests
* RPC is another way to communicate between clients and servers
* It is used commonly *within* an organisation, where it has advantages around types, performance, and streaming
* gRPC is not used to communicate from web browsers to servers (important because this is context they have)

I think your other points are good, around resource vs procedure and external APIs.